pageserver/src/tenant/timeline.rs

```diff
@@ -1659,6 +1659,10 @@ impl Timeline {
             .map(|l| (l.filename(), l))
             .collect::<HashMap<_, _>>();
 
+        // Branches might have no layers if no writes happened after the creation.
+        // Need to upload their metadata anyway, to avoid data loss due to GC.
+        let should_upload_branch_metadata = local_layers.is_empty() && index_part.is_none();
```

Comment:
This is for the case when upload was interrupted, right? Lets describe + link to corresponding issue